src/AspNetCore/Src/applicationinfo.cpp

```diff
@@ -290,13 +290,15 @@ APPLICATION_INFO::FindNativeAssemblyFromGlobalLocation(STRU* struFilename)
     DWORD dwSize = MAX_PATH;
     BOOL  fDone = FALSE;
     DWORD dwPosition = 0;
+    HMODULE hModuleANCM;
 
     // Though we could call LoadLibrary(L"aspnetcorerh.dll") relying the OS to solve
     // the path (the targeted dll is the same folder of w3wp.exe/iisexpress)
     // let's still load with full path to avoid security issue
     while (!fDone)
     {
-        DWORD dwReturnedSize = GetModuleFileName(NULL, struFilename->QueryStr(), dwSize);
+        hModuleANCM = LoadLibrary(L"aspnetcore.dll");
```

Comment:
this may not work for site extension scenario

Reply:
What would we do instead for the site extension scenario? The previous solution would never work with the site extension scenario?